### PR TITLE
Implement preset-modules for modern

### PR DESCRIPTION
**Pull request by JoviDeCroock**
`preset-env` is currently in a state where modern isn't correctly eliminating certain polyfills

Link: https://github.com/JoviDeCroock/hooked-form/issues/84

### CODE REVIEW

File: src/lib/babel-custom.js [hunk 1]
```diff
@@ -96,6 +96,10 @@ export default babelPlugin.custom(babelCore => {
 				preset.file.request.includes('@babel/preset-env'),
 			);
 
+			const environmentPreset = customOptions.modern
+				? '@babel/preset-modules'
+				: '@babel/preset-env';
+
```
Comment (developit) [inline src/lib/babel-custom.js]: I think we also need to convert `@babel/preset-env` to use `environmentPreset` on L107:

https://github.com/developit/microbundle/pull/557/files#diff-de034110aceba806ff222e668bb57fc1R107

File: src/lib/babel-custom.js [hunk 2]
```diff
@@ -93,14 +93,18 @@ export default babelPlugin.custom(babelCore => {
 			const babelOptions = config.options || {};
 
 			const envIdx = (babelOptions.presets || []).findIndex(preset =>
-				preset.file.request.includes('@babel/preset-env'),
+				preset.file.request.test(/@babel\/(preset-)?env/),
```
Comment (ForsakenHarmony) [inline src/lib/babel-custom.js]: this seems to be broken

Comment (JoviDeCroock) [reply src/lib/babel-custom.js]: Hmm, I wonder if the previous would've been better

**Comment by ForsakenHarmony** [comment]
now the snapshots seem to be a bit out of order or smth

**Comment by JoviDeCroock** [comment]
I'll run the `--updateSnapshot` again, could it be windows-related?

File: test/__snapshots__/index.test.js.snap [hunk 3]
```diff
@@ -1631,14 +1635,14 @@ modern
 
 
 Build \\"modernLib\\" to dist:
-159 B: modern-lib.modern.js.gz
```
Comment (JoviDeCroock) [inline test/__snapshots__/index.test.js.snap]: The difference in size is noticeable

Comment (developit) [reply test/__snapshots__/index.test.js.snap]: (this one mostly comes down to rest parameters being supported in preset-modules but transpiled to arguments access in preset-env)

**Pull request merged**